python_src/src/lamp_py/migrations/versions/performance_manager/695ef31623ae_rt_trunk_id_and_route_id_processing.py

```diff
@@ -0,0 +1,308 @@
+"""RT trunk_id and route_id processing
+
+Revision ID: 695ef31623ae
+Revises: de55dc40315d
+Create Date: 2023-05-05 08:20:37.529875
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = "695ef31623ae"
+down_revision = "de55dc40315d"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    op.add_column(
+        "vehicle_trips",
+        sa.Column("branch_route_id", sa.String(60), nullable=True),
+    )
+    create_rt_braintree_function = """
+        CREATE OR REPLACE FUNCTION rt_red_is_braintree_branch(p_trip_hash bytea) RETURNS boolean AS $$ 
+        DECLARE
+            found_check bool;
+        BEGIN 
+            SELECT
+                true
+            INTO
+                found_check
+            FROM 
+                vehicle_events ve
+            WHERE 
+                ve.trip_hash = p_trip_hash
+                AND ve.stop_id IN ('70097', '70098', '70099', '70100', '70101', '70102', '70103', '70104', '70105')
+            LIMIT 1
+            ;
+            IF FOUND THEN
+                RETURN true;
+            ELSE
+                RETURN false;
+            END IF;
+        END;
```

Comment:
New function algo for determining Red line `branch_route_id`. Uses postgress `FOUND` var with a select that's limited to 1 return value, should provide a performance improvement.